diff -uNr monodevelop-4.2.5/Main.sln monodevelop-4.2.5.patched/Main.sln
--- monodevelop-4.2.5/Main.sln	2014-05-26 10:06:43.000000000 +0200
+++ monodevelop-4.2.5.patched/Main.sln	2014-05-27 06:49:06.585374001 +0200
@@ -260,12 +260,6 @@
 EndProject
 Project("{FAE04EC0-301F-11D3-BF4B-00C04F79EFBC}") = "MacPlatform.Tests", "tests\MacPlatform.Tests\MacPlatform.Tests.csproj", "{C05AAB7A-523F-4109-966B-6675BE0975BA}"
 EndProject
-Project("{FAE04EC0-301F-11D3-BF4B-00C04F79EFBC}") = "TestRunner", "tests\TestRunner\TestRunner.csproj", "{EBFC7F90-C2E5-4A4C-A327-E35021BEC181}"
-EndProject
-Project("{FAE04EC0-301F-11D3-BF4B-00C04F79EFBC}") = "MonoDevelop.TextEditor.Tests", "src\core\MonoDevelop.TextEditor.Tests\MonoDevelop.TextEditor.Tests.csproj", "{9E4BA410-8338-42EC-AF9C-422C35ECED81}"
-EndProject
-Project("{FAE04EC0-301F-11D3-BF4B-00C04F79EFBC}") = "Ide.Tests", "tests\Ide.Tests\Ide.Tests.csproj", "{73D4CC8B-BAB9-4A29-841B-F25C6311F067}"
-EndProject
 Global
 	GlobalSection(SolutionConfigurationPlatforms) = preSolution
 		Debug|Any CPU = Debug|Any CPU
diff -uNr monodevelop-4.2.5/Makefile.in monodevelop-4.2.5.patched/Makefile.in
--- monodevelop-4.2.5/Makefile.in	2014-05-26 10:06:46.000000000 +0200
+++ monodevelop-4.2.5.patched/Makefile.in	2014-05-27 07:02:54.717373954 +0200
@@ -1102,9 +1102,10 @@
 
 $(buildinfo):
 	@mkdir -p "$(top_builddir)/build/bin"
-	@$(MD_CONFIGURE) gen-buildinfo "$(top_builddir)/build/bin"
+	#@$(MD_CONFIGURE) gen-buildinfo "$(top_builddir)/build/bin"
 	@echo "Updated build information"
-	@cat $(buildinfo)
+	#@cat $(buildinfo)
+	echo "echo 'not implemented';" >  "$(top_builddir)/build/bin/buildinfo"
 
 buildinfo: $(buildinfo)
 
diff -uNr monodevelop-4.2.5/src/core/MonoDevelop.Core/BuildVariables.cs.in monodevelop-4.2.5.patched/src/core/MonoDevelop.Core/BuildVariables.cs.in
--- monodevelop-4.2.5/src/core/MonoDevelop.Core/BuildVariables.cs.in	1970-01-01 01:00:00.000000000 +0100
+++ monodevelop-4.2.5.patched/src/core/MonoDevelop.Core/BuildVariables.cs.in	2014-05-27 06:45:45.257374329 +0200
@@ -0,0 +1,9 @@
+namespace MonoDevelop
+{
+	public static class BuildInfo
+	{
+		public const string Version = "@PACKAGE_VERSION@";
+		public const string VersionLabel = "@PACKAGE_VERSION_LABEL@";
+		public const string CompatVersion = "@COMPAT_ADDIN_VERSION@";
+	}
+}
diff -uNr monodevelop-4.2.5/src/core/MonoDevelop.Core/BuildVariables.gen.cs monodevelop-4.2.5.patched/src/core/MonoDevelop.Core/BuildVariables.gen.cs
--- monodevelop-4.2.5/src/core/MonoDevelop.Core/BuildVariables.gen.cs	2014-05-26 10:06:42.000000000 +0200
+++ monodevelop-4.2.5.patched/src/core/MonoDevelop.Core/BuildVariables.gen.cs	2014-05-27 06:44:46.713374002 +0200
@@ -9,7 +9,7 @@
 		public static void Main (string[] args)
 		{
 			var dir = args [0];
-			var lines = File.ReadAllLines (Path.Combine (dir, "..", "..", "..", "..", "version.config"));
+			var lines = File.ReadAllLines (Path.Combine (dir, "..", "..", "..", "version.config"));
 			var txt = File.ReadAllText (Path.Combine (dir, "BuildVariables.cs.in"));
 			txt = txt.Replace ("@PACKAGE_VERSION@", GetValue (lines, "Version"));
 			txt = txt.Replace ("@PACKAGE_VERSION_LABEL@", GetValue (lines, "Label"));
diff -uNr monodevelop-4.2.5/src/core/MonoDevelop.Ide/gtkrc monodevelop-4.2.5.patched/src/core/MonoDevelop.Ide/gtkrc
--- monodevelop-4.2.5/src/core/MonoDevelop.Ide/gtkrc	1970-01-01 01:00:00.000000000 +0100
+++ monodevelop-4.2.5.patched/src/core/MonoDevelop.Ide/gtkrc	2014-05-27 06:58:51.853376091 +0200
@@ -0,0 +1,387 @@
+# Theme for Xamarin composed by Lanedo GmbH
+# Copyright 2012 Xamarin Inc.
+# Authors: 
+#   Christian Kellner <christian.kellner@lanedo.com>
+#   Carlos Garnacho <carlos.garnacho@lanedo.com>
+
+gtk-color-scheme =
+"bg_color:         #f5f5f5
+fg_color:          #000
+base_color:        #fff
+text_color:        #000
+selected_bg_color: #649dd6
+selected_fg_color: #fff
+tooltip_bg_color:  #fff9e5
+tooltip_fg_color:  #000"
+
+gtk-button-images     = 0
+gtk-menu-images       = 0
+gtk-toolbar-style     = 0
+gtk-enable-mnemonics  = 0
+gtk-icon-sizes        = "gtk-small-toolbar=16,16:gtk-large-toolbar=22,22"
+gtk-toolbar-icon-size = large-toolbar
+gtk-auto-mnemonics    = 1
+
+
+style "default" {
+    xthickness = 1
+    ythickness = 1
+
+    ### colours
+    bg[NORMAL]        = @bg_color
+    bg[PRELIGHT]      = @bg_color
+    bg[SELECTED]      = @selected_bg_color
+    bg[INSENSITIVE]   = @bg_color
+    bg[ACTIVE]        = @bg_color
+
+    fg[NORMAL]        = @fg_color
+    fg[PRELIGHT]      = @fg_color
+    fg[SELECTED]      = @selected_fg_color
+    fg[INSENSITIVE]   = darker (@bg_color)
+    fg[ACTIVE]        = @fg_color
+
+    text[NORMAL]      = @text_color
+    text[PRELIGHT]    = @text_color
+    text[SELECTED]    = @selected_fg_color
+    text[INSENSITIVE] = darker (@bg_color)
+    text[ACTIVE]      = @text_color
+
+    base[NORMAL]      = @base_color
+    base[PRELIGHT]    = @selected_bg_color
+    base[SELECTED]    = @selected_bg_color
+    base[INSENSITIVE] = @bg_color
+    base[ACTIVE]      = shade (1.3, @selected_bg_color)
+
+    ### style properties
+    GtkWidget::new-tooltip-style = 1
+
+    GtkEntry::honors-transparent-bg-hint = 1
+    GtkEntry::inner-border = { 2, 2, 2, 2 }
+
+    GtkButton::child-displacement-x = 0
+    GtkButton::child-displacement-y = 0
+    GtkButton::default-border = { 0, 0, 0, 0 }
+    GtkButton::inner-border = { 2, 2, 4, 4 }
+
+    GtkTreeView::odd-row-color = @base_color
+
+    GtkScrolledWindow::scrollbar-spacing = 0
+    GtkScale::slider-length = 14
+    GtkRange::trough-side-details = 1
+
+    GtkTreeView::expander-size = 8
+    GtkExpander::expander-size = 8
+    GtkComboBox::arrow-size = 12
+
+    engine "xamarin" {
+
+        arrowstyle = 2
+        rgba       = TRUE
+        roundness  = 3
+        glazestyle = 0
+
+        menustyle        = 0
+        menuitemstyle    = 0
+        menubaritemstyle = 0
+        comboboxstyle    = 1
+        spinbuttonstyle  = 1
+        colorize_scrollbar = FALSE
+        progressbarstyle = 0
+        trough_shades = { 0.842, 0.886 }
+    }
+}
+
+style "wide" = "default" {
+    xthickness = 2
+    ythickness = 2
+}
+
+style "wider" = "default" {
+    xthickness = 3
+    ythickness = 3
+}
+
+
+style "button" {
+    xthickness = 2
+    ythickness = 4
+
+    bg[NORMAL]      = @bg_color
+    bg[PRELIGHT]    = shade(1.1, @bg_color)
+    bg[ACTIVE]      = @bg_color
+    bg[INSENSITIVE] = @bg_color
+    GtkWidget::focus-padding = 0
+
+    engine "xamarin" {
+        contrast = 1.0
+        gradient_shades     = {0.96, 0.918, 0.926, 0.858}
+        focus_color         = shade(0.6, @bg_color)
+        textstyle           = 1
+      }
+}
+
+style "toggle-button" = "button" {
+    bg[NORMAL] = @base_color
+    bg[PRELIGHT] = shade(0.98, @base_color)
+}
+
+style "menu" {
+    xthickness = 0
+    ythickness = 0
+
+    fg[PRELIGHT] = @selected_fg_color
+
+    engine "xamarin" {
+        roundness = 3
+        gradient_shades     = {1.02, 0.97, 0.92, 0.87}
+    }
+}
+
+style "menu-item" {
+    xthickness = 0
+    ythickness = 3
+
+    fg[PRELIGHT] = @selected_fg_color
+
+    engine "xamarin" {
+        roundness = 0
+        gradient_shades     = {1.02, 0.97, 0.92, 0.87}
+    }
+}
+
+style "toolbar" = "default"
+{
+    bg[NORMAL] = "#ddd"
+    engine "xamarin" {
+        toolbarstyle = 2
+    }
+}
+
+style "toolbar-button"
+{
+    engine "xamarin" {
+        contrast = 1.0
+        focus_color         = @bg_color
+      }
+}
+
+style "tooltip"
+{
+    xthickness = 5
+    ythickness = 5
+
+    fg[NORMAL] = @tooltip_fg_color
+    bg[NORMAL] = @tooltip_bg_color
+}
+
+style "treeview" = "default"
+{
+    GtkTreeView::odd-row-color = "#f5f5f5"
+
+    engine "xamarin" {
+        roundness = 0
+        gradient_shades     = {1.0, 1.0, 1.0, 1.0}
+    }
+}
+
+style "tree-header" = "wide"
+{
+    bg[NORMAL] = "#f2f2f2"
+
+    engine "xamarin" {
+        listviewheaderstyle = 3
+        separatorstyle = 1
+    }
+}
+
+style "icon-view" = "default"
+{
+    bg[SELECTED] = "#f5f7fa"
+    base[SELECTED] = "#f5f7fa"
+    fg[SELECTED] = "#000"
+    text[SELECTED] = "#000"
+}
+
+style "toolbar-combo-box" = "default"
+{
+    fg[NORMAL] = "#616161"
+    text[NORMAL] = @fg_color
+    bg[NORMAL] = "#fff"
+    base[NORMAL] = "#fff"
+    ythickness = 4
+    xthickness = 2
+
+    engine "xamarin" {
+	comboboxstyle = 0
+        arrowstyle = 2
+    }
+}
+
+style "combo-box" = "default"
+{
+    fg[NORMAL] = "#616161"
+    xthickness = 2
+    ythickness = 4
+
+    engine "xamarin" {
+        arrowstyle = 4
+    }
+}
+
+style "combo-box-entry" = "combo-box"
+{
+    engine "xamarin" {
+        arrowstyle = 3
+    }
+}
+
+style "combo-box-label" = "combo-box"
+{
+    fg[NORMAL] = "#595959"
+}
+
+style "notebook" = "default"
+{
+    GtkNotebook::tab-overlap = 1
+    bg[ACTIVE] = @base_color
+    bg[NORMAL] = @base_color
+
+    engine "xamarin" {
+        gradient_shades = { 0.945, 0.945, 0.945, 0.685 }
+    }
+}
+
+style "radio-button" = "default"
+{
+    GtkCheckButton::indicator-size = 15
+    text[NORMAL] = lighter (@text_color)
+    base[NORMAL] = @base_color
+    text[ACTIVE] = lighter(@text_color)
+}
+
+style "check-box" = "default"
+{
+    GtkCheckButton::indicator-size = 15
+    base[NORMAL] = "#f8f8f8"
+    text[NORMAL] = "#404040"
+    text[ACTIVE] = "#404040"
+}
+
+style "entry" = "default"
+{
+    xthickness = 4
+    ythickness = 4
+    GtkEntry::inner-border = { 2, 2, 2, 2 }
+    base[INSENSITIVE] = shade(1.02, @bg_color)
+}
+
+style "search-entry" = "wider"
+{
+    xthickness = 6
+    ythickness = 4
+    bg[NORMAL] = @base_color
+}
+
+style "scrollbar" = "default"
+{
+    GtkScrollbar::has-forward-stepper = 0
+    GtkScrollbar::has-backward-stepper = 0
+    GtkRange::slider-width = 10
+    GtkRange::trough-border = 2
+    GtkRange::trough-side-details = 0
+
+    bg[SELECTED] = @bg_color
+    bg[NORMAL] = @bg_color
+
+    engine "xamarin" {
+        trough_shades = { 0.968, 1.04 }
+        roundness = 0
+    }
+}
+
+style "progressbar" = "default"
+{
+    bg[SELECTED] = "#62b3ec"
+    fg[SELECTED] = "#3a67a6"
+    xthickness = 0
+    ythickness = 0
+}
+
+style "scrolled-window" = "default"
+{
+    xthickness = 1
+    ythickness = 1
+}
+
+style "scale" = "default"
+{
+    bg[SELECTED] = @selected_bg_color
+
+    engine "xamarin" {
+        roundness = 7
+    }
+}
+
+style "spin-button" = "wider"
+{
+    fg[NORMAL] = "#616161"
+    ythickness = 4
+    GtkEntry::inner-border = { 2, 2, 1, 1 }
+    engine "xamarin" {
+        arrowstyle = 3
+    }
+}
+
+### Apply the styles
+class "GtkWidget"    style "default"
+
+class "GtkEntry"     style "entry"
+class "GtkSpinButton" style "spin-button"
+class "GtkFrame"     style "wider"
+class "GtkRange"     style "wide"
+class "GtkSeparator" style "wide"
+class "GtkScrollbar" style "scrollbar"
+class "GtkProgressBar" style "progressbar"
+class "GtkScrolledWindow" style "scrolled-window"
+class "GtkNotebook" style "notebook"
+class "GtkButton" style "button"
+
+class "GtkRadioButton" style "radio-button"
+class "GtkCheckButton" style "check-box"
+class "GtkScale" style "scale"
+
+#widget "*GtkEntry" style "entry"
+widget "*search-entry*" style "search-entry"
+widget "*GtkTextView"   style "entry"
+
+widget_class "*<GtkButton>"     style "button"
+widget_class "*<GtkToggleButton>" style "toggle-button"
+
+widget_class "*<GtkMenu>*"      style "menu"
+widget_class "*<GtkMenuItem>*"  style "menu-item"
+
+widget_class "*<GtkToolbar>*" style "toolbar"
+widget_class "*ToolButton*"     style "toolbar-button"
+
+widget_class "*.<GtkTreeView>*"         style "treeview"
+widget_class "*.GtkTreeView.GtkButton"  style "tree-header"
+widget_class "*.GtkList.GtkButton"  style "tree-header"
+
+widget_class "*<GtkIconView>" style "icon-view"
+
+widget_class "*.<GtkComboBox>.*" style "combo-box"
+widget_class "*.<GtkComboBoxText>.*" style "combo-box-entry"
+widget_class "*.<GtkComboBoxEntry>.*" style "combo-box-entry"
+widget_class "*.<GtkComboBoxText>.<GtkEntry>" style "entry"
+widget_class "*.<GtkComboBoxEntry>.<GtkEntry>" style "entry"
+widget_class "*.<GtkComboBoxText>.<GtkToggleButton>" style "button"
+widget_class "*.<GtkComboBoxEntry>.<GtkToggleButton>" style "button"
+#widget_class "*.<GtkComboBoxEntry>.GtkEntry" style "entry"
+#widget_class "*.<GtkComboBoxEntry>.<GtkButton>" style "button"
+widget_class "*.<GtkComboBox>.*.GtkLabel" style "combo-box-label"
+
+# Comboboxes within toolbars
+widget_class "*.<GtkToolbar>.*.<GtkComboBox>.*" style "toolbar-combo-box"
+widget "*MainToolbar*.GtkComboBox.*" style "toolbar-combo-box"
+
+widget "gtk-tooltip*"       style "tooltip"
diff -uNr monodevelop-4.2.5/src/core/MonoDevelop.Ide/MonoDevelop.Ide.csproj monodevelop-4.2.5.patched/src/core/MonoDevelop.Ide/MonoDevelop.Ide.csproj
--- monodevelop-4.2.5/src/core/MonoDevelop.Ide/MonoDevelop.Ide.csproj	2014-05-26 10:06:43.000000000 +0200
+++ monodevelop-4.2.5.patched/src/core/MonoDevelop.Ide/MonoDevelop.Ide.csproj	2014-05-27 06:59:45.353369268 +0200
@@ -1951,12 +1951,6 @@
     <Folder Include="MonoDevelop.Ide.TextEditing\" />
   </ItemGroup>
   <ItemGroup>
-    <Content Include="gtkrc.win32">
-      <CopyToOutputDirectory>PreserveNewest</CopyToOutputDirectory>
-    </Content>
-    <Content Include="gtkrc.mac">
-      <CopyToOutputDirectory>PreserveNewest</CopyToOutputDirectory>
-    </Content>
     <Content Include="gtkrc">
       <CopyToOutputDirectory>PreserveNewest</CopyToOutputDirectory>
     </Content>
diff -uNr monodevelop-4.2.5/version.config monodevelop-4.2.5.patched/version.config
--- monodevelop-4.2.5/version.config	1970-01-01 01:00:00.000000000 +0100
+++ monodevelop-4.2.5.patched/version.config	2014-05-27 06:46:06.949355687 +0200
@@ -0,0 +1,3 @@
+Version=4.2.5
+Label=4.2.5
+CompatVersion=4.0
