--- monodevelop-4.2.5/Makefile.in.orig	2014-05-24 20:44:15.240719181 +0200
+++ monodevelop-4.2.5/Makefile.in	2014-05-24 20:44:43.856715277 +0200
@@ -1072,10 +1072,10 @@
 	touch vcrevision
 
 $(buildinfo):
-	@mkdir -p "$(top_builddir)/build/bin"
-	@$(MD_CONFIGURE) gen-buildinfo "$(top_builddir)/build/bin"
-	@echo "Updated build information"
-	@cat $(buildinfo)
+	#@mkdir -p "$(top_builddir)/build/bin"
+	#@$(MD_CONFIGURE) gen-buildinfo "$(top_builddir)/build/bin"
+	#@echo "Updated build information"
+	#@cat $(buildinfo)
 
 buildinfo: $(buildinfo)
 
