--- a/Makefile	2016-08-12 21:54:17.077712312 +0200
+++ b/Makefile	2016-08-12 21:54:54.754660209 +0200
@@ -91,11 +91,11 @@
 	@cd main && make buildinfo
 	@cp main/build/bin/buildinfo tarballs/monodevelop-$(PACKAGE_VERSION)/
 	@echo Generating merged tarball
-	@find tarballs/monodevelop-$(PACKAGE_VERSION)/ -type f -a \
-		\( -name \*.exe -o \
-		-name \*.dll -o \
-		-name \*.mdb \) \
-		-delete
+	#@find tarballs/monodevelop-$(PACKAGE_VERSION)/ -type f -a \
+	#	\( -name \*.exe -o \
+	#	-name \*.dll -o \
+	#	-name \*.mdb \) \
+	#	-delete
 	@cd tarballs && tar -cjf monodevelop-$(PACKAGE_VERSION).tar.bz2 monodevelop-$(PACKAGE_VERSION)
 	@cd tarballs && rm -rf monodevelop-$(PACKAGE_VERSION)
 
