--- a/mcs/class/System/System.Net/HttpWebRequest.cs	2013-02-27 17:59:04.000000000 +0100
+++ b/mcs/class/System/System.Net/HttpWebRequest.cs	2013-05-04 07:08:23.497720746 +0200
@@ -979,8 +979,12 @@
 		}
 #endif
 
+		// see https://bugzilla.xamarin.com/show_bug.cgi?id=11833
+		internal bool asynchronous = true;
+
 		public override WebResponse GetResponse()
 		{
+			asynchronous = false;
 			WebAsyncResult result = (WebAsyncResult) BeginGetResponse (null, null);
 			return EndGetResponse (result);
 		}
--- a/mcs/class/System/System.Net/WebConnection.cs	2013-02-27 17:59:04.000000000 +0100
+++ b/mcs/class/System/System.Net/WebConnection.cs	2013-05-04 07:09:16.930719147 +0200
@@ -811,7 +811,11 @@
 			lock (this) {
 				if (state.TrySetBusy ()) {
 					status = WebExceptionStatus.Success;
-					ThreadPool.QueueUserWorkItem (initConn, request);
+// see https://bugzilla.xamarin.com/show_bug.cgi?id=11833
+if (request.asynchronous)
+     ThreadPool.QueueUserWorkItem (initConn, request);
+else
+     initConn.Invoke (request);
 				} else {
 					lock (queue) {
 #if MONOTOUCH
