--- a/mono/metadata/gc.c	2015-01-14 22:40:45.146934793 +0100
+++ b/mono/metadata/gc.c	2015-01-14 22:41:17.032452339 +0100
@@ -1224,7 +1224,7 @@
 				ret = WaitForSingleObjectEx (gc_thread->handle, INFINITE, TRUE);
 				g_assert (ret == WAIT_OBJECT_0);
 
-				mono_thread_join (MONO_UINT_TO_NATIVE_THREAD_ID (gc_thread->tid));
+				mono_thread_join ((gpointer)MONO_UINT_TO_NATIVE_THREAD_ID (gc_thread->tid));
 			}
 		}
 		gc_thread = NULL;

